src/app/components/button/button.component.ts

```diff
@@ -0,0 +1,29 @@
+import { Component, Input} from '@angular/core';
+
+/**
+ * ButtonComponent
+ * Componente dinâmico de botão customizado.
+ * 
+ * @param texto - Texto que será exibido no botão
+ * @param id - Id do botão
+ * @param class - classe aplicada ao botão, que resultará na criação de um elemento <a> ou <button> e estilização diferente no css
+ * @param routerLink - Rota para navegação, ao clicar no botão
+ * @param type - Tipo do botão
+ * 
+ */
+
+
+@Component({
+  selector: 'app-button',
+  templateUrl: './button.component.html',
+  styleUrls: ['./button.component.scss']
+})
+export class ButtonComponent {
+  @Input() texto = '';
+  @Input() id = '';
+  @Input() class ='';
```

Comment:
Sugestão: Em vez de class, e se for chamado de type? ai pode ser link ou button, o que acha?

Reply:
Eu estou usando o type quando o botão está em um formulário, nas outras telas, pra passar a informação submit. Essa classe fica na div e está controlando a estilização, além de verificar se será criado um elemento a ou button. Como ficaria se eu trocasse?